src/donut/dbxray/generate/datapotato.cljc

```diff
@@ -0,0 +1,43 @@
+(ns donut.dbxray.generate.datapotato
+  (:require
+   [clojure.string :as string]
+   [donut.dbxray.generate :as ddg]))
+
+(defn- gen-table-prefix
+  [table-name]
+  (let [words (string/split (name table-name) #"_")
+        initials (map (comp str first) words)]
+    {:prefix (keyword (apply str initials))}))
+
+(defn- add-table-prefix-to-col-name
+  [table-name col-name]
+  (let [table (name table-name)]
+    (keyword (str table "/" (name col-name)))))
+
+(defn- gen-relations-for-column
+  [table-name column-name column-data]
+  (when-let [refers-to (:refers-to column-data)]
+    (let [relations-key (add-table-prefix-to-col-name table-name column-name)
+          [ref-table ref-col] refers-to
+          relations-path [ref-table (add-table-prefix-to-col-name ref-table ref-col)]]
+      {:relations {relations-key relations-path}})))
+
+(defn- gen-potato-for-table
+  [table-name table-data]
+  (let [{:keys [columns]} table-data
+        prefix (gen-table-prefix table-name)]
+    {table-name (merge prefix
+                       (reduce (fn [relations [column-name column-data]]
+                                 (merge-with into
+                                             relations
+                                             (gen-relations-for-column table-name column-name column-data)))
+                               {}
+                               columns))}))
+
+
+(defn generate
```

Comment:
I'd like to use `flatland.ordered.map/ordered-map` here. The output of `donut.dbxray/xray` is an ordered map where the keys (table names) are topologically sorted by foreign key dependencies. I'd like that ordering to be preserved in the potato schema that `generate` returns.

I _think_ it should be possible to do this by changing the second argument to reduce here to be `(omap/ordered-map)` instead of `{}`.

Another possibility would be to use `(into (omap/ordered-map) (map gen-potato-for-table))`, which I think would require updating `gen-potato-for-table` to return vectors instead of maps.

Reply:
this is where the tables are ordered: https://github.com/donut-party/dbxray/blob/main/src/donut/dbxray.clj#L180

I don't think it's necessary to put the table relations in order

Reply:
ok, I've done that and updated the  test to check for order.

Also updated the `fixture` data to reflect the intended top-level ordering: https://github.com/donut-party/dbxray/commit/c376e78901f3a27300faa4498f6f9439112fa267